core/src/main/java/ai/timefold/solver/core/config/score/director/ConstraintProfilingMode.java

```diff
@@ -0,0 +1,22 @@
+package ai.timefold.solver.core.config.score.director;
+
+/**
+ * Controls how constraints are profiled.
+ * Enabling profiling have a minor performance impact.
+ */
+public enum ConstraintProfilingMode {
```

Comment:
IMO this is needless complexity. Do not profile by method or line - profile _by constraint_. That is what people actually want to see; the list of constraints and how much time they take to run.

Also, the names of constraints typically do not change, while method names and lines do while code is refactored over time. This should IMO not be information we depend on, or display.